public/app/plugins/datasource/opentsdb/migrations.ts

```diff
@@ -0,0 +1,28 @@
+type LegacyAnnotation = {
+  fromAnnotations?: boolean;
+  isGlobal?: boolean;
+  target?: string
+  name?: string;
+};
+
+// this becomes the target in the migrated annotations
+const migrateLegacyAnnotation = (json: LegacyAnnotation) => {
+  // return the target annotation
+  let annotation: LegacyAnnotation = {
+    fromAnnotations: true,
+    target: json.target ?? '',
+    name: json.name ?? '',
+    isGlobal: json.isGlobal ?? false,
+  };
+
+  return annotation;
+};
+
+// eslint-ignore-next-line
+export const prepareAnnotation = (json: any) => {
+  const resultingTarget = json.target && typeof json.target !== 'string' ? json.target : migrateLegacyAnnotation(json);
+
+  json.target = resultingTarget;
```

Comment:
It would be nice not to change the object being passed as a parameter. You could maybe have something like:

```javascript
export const prepareAnnotation = (json: any) => {
  const target = json.target && typeof json.target !== 'string' ? json.target : migrateLegacyAnnotation(json);

  return { ...json, target };
};
```
What do you think?

Reply:
I was thinking this would work, but I found that the code I'm using to migrate the annotations looks to see if the annotations object has changed, then runs an onChange callback and ends up preparing the again, looping forever. 
 https://github.com/grafana/grafana/blob/4ca845e47906bae299eaf4bcf3ecd78cf1f0f2c5/public/app/features/annotations/components/StandardAnnotationQueryEditor.tsx#L50-L56